packages/core/src/components/callout/callout.tsx

```diff
@@ -7,12 +7,16 @@
 import * as classNames from "classnames";
 import * as React from "react";
 
-import { Classes, IIntentProps, IProps } from "../../common";
+import { Classes, IIntentProps, Intent, IProps } from "../../common";
+import { Icon } from "../../index";
 import { IconName } from "../icon/icon";
 
 /** This component also supports the full range of HTML `<div>` props. */
 export interface ICalloutProps extends IIntentProps, IProps {
-    /** Name of icon to render on left-hand side. */
+    /**
+     * Name of icon to render on left-hand side.
+     * If this prop is omitted, the `intent` prop will determine a default icon.
```

Comment:
are you sure about this? I'll bet there are designs that call for a colored callout without an icon

Reply:
`iconName={null}` should do it

Reply:
then you need `iconName?: IconName | null;` below for strict null checks. also documentation.